docs/guides/setting-up.md

```diff
@@ -0,0 +1,118 @@
+
+## Add necessary kernel paramaters
+
+Using your bootloader, add the `intel_iommu=on iommu=pt pcie_ports=compat` kernel parameters. For example in GRUB :-
+
+  1. Edit `/etc/default/grub`.
+  2. On the line with `GRUB_CMDLINE_LINUX="quiet splash"`, add the following kernel parameters: `intel_iommu=on iommu=pt pcie_ports=compat`.
+  3. Run `sudo grub-mkconfig -o /boot/grub/grub.cfg` if you are on a non-debian based distro. If using Debian or Ubuntu based distro, run `sudo update-grub`.
+
+## Make modules load on boot
+
+Simply run the following :-
+
+```sh
+echo apple-bce | sudo tee /etc/modules-load.d/t2.conf
+```
+
+## Make modules load on early boot
+
+Having the `apple-bce` module loaded early allows the use of the keyboard for decrypting encrypted volumes (LUKS).
+It also is useful when boot doesn't work, and the keyboard is required for debugging.
+To do this, one must ensure the `apple-bce` module *as well as its dependent modules* are included in the initial ram disk.
+If your distro uses `initramfs-tools` (all debian-based distros), then `/etc/initramfs-tools/modules` stores a list of extra modules to be included and loaded at early boot time.
+Run `sudo su` to open a shell as root and run the following over there:
+
+```sh
+cat <<EOF >> /etc/initramfs-tools/modules
+# Required modules for getting the built-in apple keyboard to work:
+snd
+snd_pcm
+apple-bce
+EOF
+update-initramfs -u
+```
+
+Other distros use a different initramfs module loading mechanism.
+For example in Arch ensure that the `/etc/mkinitcpio.conf` file has:
+
+```sh
+MODULES="apple-bce"
+```
+
+And then run `sudo mkinitcpio -P`.
+See your distro-specific instructions for configuring `apple-bce` to added to your initramfs.
+
+## Setting up the Touch Bar
+
+The Touch Bar can be set up by running [this script](../tools/touchbar.sh) **in Linux** using `bash /path/to/script`. Make sure your Linux kernel and macOS is updated before running this script.
+
+After running this script, if you wish to change the default mode of the Touch Bar, run `sudo touchbar` and choose the mode you wish.
+
+In case your Touch Bar is unable to change modes on pressing the fn key, you could try the following :-
+
+- Try adding `usbhid.quirks=0x05ac:0x8302:0x80000` as a Kernel Parameter using your Bootloader.
+- Try running the following and rebooting.
+  
+   ```sh
+   echo -e "# delay loading of the touchbar driver\ninstall apple-ib-tb /bin/sleep 7; /sbin/modprobe --ignore-install apple-ib-tb" | sudo tee /etc/modprobe.d/delay-tb.conf >/dev/null
+   ```
+  
+- Boot into the [macOS Recovery](https://support.apple.com/en-gb/HT201314) and then restart into Linux.
+- Unplug all the external USB keyboards and mouse and then restart into Linux, keeping them unplugged.
+
+If you still face an issue, mention it [here](https://github.com/t2linux/wiki/issues) or on the discord.
+
+## Fixing Suspend
+
+Copy [this script](../tools/rmmod_tb.sh) to `/lib/systemd/system-sleep/rmmod_tb.sh`
```

Comment:
if the script is not already there I suppose?

Reply:
Wdym?

Reply:
iirc, some t2 specific iso's put that script there already? although it might be an outdated copy of it

Reply:
I don’t think it causes any functional change if the user follows it again. Moreover the guide initially has a Do I need to follow this section.